### PR TITLE
Make it easier to run verification test locally

### PR DESCRIPTION
**Pull request by steiza**
For #99

#### Summary

If you're only testing verification use-cases locally, you don't need the OIDC JWT.

You can test this change with:

```sh
(env) $ pytest test --skip-signing --entrypoint=SIGSTORE_CLIENT
```

#### Release Note

* Enabled easier local testing for verification with `pytest test --skip-signing --entrypoint=SIGSTORE_CLIENT`

#### Documentation

N/A

### CODE REVIEW

File: test/conftest.py [hunk 1]
```diff
@@ -52,7 +52,6 @@ def pytest_addoption(parser) -> None:
         "--github-token",
         action="store",
         help="the GitHub token to supply to the Sigstore client under test",
-        required=True,
```
Comment (steiza) [inline test/conftest.py:55]: This is maybe not idea - now if you leave off `--github-token` (and didn't specify `--skip-staging`) you'll get a bunch of `401 Client Error: Unauthorized for url: https://api.github.com/repos/sigstore-conformance/extremely-dangerous-public-oidc-beacon/actions/workflows/...` errors.

Comment (woodruffw) [reply test/conftest.py:55]: Hmm, perhaps we could test for `--skip-staging` and, if not present, make the absence of `--github-token` a hard error? That would handle the use case here while also simultaneously preventing a bunch of testcases that we know will fail 🙂

**Review by woodruffw — COMMENTED** [review]
LGTM overall, one design thought!

File: test/conftest.py (hunk 1 above)
Comment (steiza) [reply test/conftest.py:55]: I think https://github.com/sigstore/sigstore-conformance/pull/100/commits/3d8279cb3459a06076788a2c9e36ea0796b140eb should address this!

**Review by woodruffw — APPROVED** [review]

**Comment by woodruffw** [comment]
Thanks @steiza!

**Pull request merged**